sys/src/9/amd64/l64vsyscall.S

```diff
@@ -73,15 +73,16 @@ syscallentry:
 	MOVQ	%rsp, %rsi 				/* Ureg* */
 	// system call number is in %rax, as per linux.
 	movq	%rax, %rdi
-	xorq	%rax, %rax
-	pushq	%rax
+	xorl	%eax, %eax
```

Comment:
why not xorq?

Reply:
It's unnecessary and a longer instruction (e.g., more instruction bytes). xorl will clear the upper 32 bits of the destination register anyway; this is the form recommended by the Intel optimization manual, which I've standardized on everywhere else as a matter of course (e.g., not just for optimized sections of code).

Reply:
fantastic.